parlai/tasks/style_gen/agents.py

```diff
@@ -88,3 +89,45 @@ def __init__(self, opt, shared=None):
             opt=opt, base_task='blended_skill_talk:WoWPersonaTopicifierTeacher'
         )
         super().__init__(opt, shared=shared)
+
+
+class PrevCurrUttStyleTeacher(AbstractWrapperTeacher):
+    """
+    Serving examples for use with projects.style_gen.classifier:ClassifierAgent.
+
+    This teacher will replace message['text'] with a concatenation of the last utterance
+    in message['text'] and message['labels'][0], and it will replace message['labels']
+    with [message['personality']]. This is to allow for training/evaluation of
+    projects.style_gen.classifier:ClassifierAgent, which typically classifies the style
+    of an utterance given that utterance and the previous one as context.
+
+    Because the dialogue history is effectively overwritten by this action, all episodes
+    will be flattened into one example each.
+    """
+
+    def __init__(self, opt: Opt, shared=None):
+        super().__init__(opt, shared)
+        assert isinstance(self.task, FixedDialogTeacher)
+
+    def act(self):
```

Comment:
this looks quite similar to the `LabelsToText` teacher in the `wrapper/agents.py` file; is the only difference that we set the label to `act['personality']`?

Reply:
Yes - and also, here we truncate the context history to only the previous utterance, which is necessary for how we trained the style-gen classifier